pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -250,14 +254,14 @@ def state_key(self) -> str:
     def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: Optional[str] = None) -> None:
         # NOTE: setting these attributes needs to happen as early as possible BEFORE reloading callback states,
         # because the attributes are part of the state_key which needs to be fully defined before reloading.
+        self.__resolve_ckpt_dir(trainer)
```

Comment:
since it is moved now, can you add a test reflecting the case that covers this case, where `dirpath` is not set explicitly, but is created dynamically and will be different from what it was during the initial run?